### DISCUSSION TITLE
pattern with overlapping shapes

### DISCUSSION THREAD

**Discussion by none511** [fabricjs/fabric.js#8496] [Q&A]
hi all!
i'm playing with patterns.. fabric gives the possibility to set scale, rotation and offset, however, i'd like to obtain something like this:
<img width="695" alt="Screenshot 2022-12-05 alle 7 28 12 PM" src="https://user-images.githubusercontent.com/42898837/205714623-a611d7d1-f7f9-475a-a2de-e73a877cd3cf.png">
actually the only idea i came up with, is to clone the svg, i think it could work but maybe there is a better and more performant way to do it.. any suggestions? 
thanx!

**Comment by ShaMan123** [comment]
Dropping some thoughts, without enough experience in patterns.
First of all there's `patternTransform` based on https://developer.mozilla.org/en-US/docs/Web/API/CanvasPattern/setTransform, look at the demo in the link, it does something very relevant to want to need. I guess if you should apply an inverted transform to each star, distorting it. Then once the pattern will be transformed as a whole, both the position of stars and their appearance will match your ref.
Another thought... I wonder if you could achieve this pattern with `repeat-x` or `repeat-y`, splitting this pattern into diagonal columns/rows respectively...

**Comment by none511** [comment]
thx shaman, i explored some funcionality of patterns, but the problem is that looks like is not possibile to overlap the tiles. as for the translate in matrix, it allows to set the offset of the whole pattern, whereas i'd like to set the offset of the tile in relation to the other.
start to think there's no way to achieve that using patterns.

**Comment by ShaMan123** [comment]
OK I have a proposal perhaps.
I am using the following for the eraser brush #8499 
- render the shape in black
- set `globalCompositeOperation: 'source-in'`
- render pattern

This way you sort of impl your own canvas pattern
It requires context isolation (so use object caching)

You could achieve this by tweaking Object's render cache methods or `drawObject` or `_render`

**Reply by ShaMan123** [reply]
And to render the star you could use a single fabric object with object caching to boost perf and move it each iteration step
